### PR TITLE
Fixed header and footer(SPA)- #1729

### PR DESCRIPTION
**Pull request by juhijariwala**

### CODE REVIEW

File: server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_navigation.html.erb [hunk 1]
```diff
@@ -1,41 +1,57 @@
...
-    <!-- Remove the class "menu-icon" to get rid of menu icon. Take out "Menu" to just have icon alone -->
+      <%= link_to '', main_app.pipeline_dashboard_path, class: "logo" -%>    </li>
     <li class="toggle-topbar menu-icon"><a href="#"><span>Menu</span></a></li>
   </ul>
 
   <section class="top-bar-section">
-    <ul class="right">
-        <li class="has-dropdown">
-            <a href="#" class="user">admin</a>
+    <% unless @no_tabs %>
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_navigation.html.erb]: Who is setting this?

File: server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_dropdown_admin.html.erb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<li role="presentation">
+  <%= link_to l.string("PIPELINES_POPUP_TITLE"), main_app.pipeline_groups_path -%>
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_dropdown_admin.html.erb]: Why is the `main_app` needed — can't we call `foo_path` directly?

File: server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_dropdown_template_admin.html.erb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<li role="presentation">
+  <%= link_to l.string("TEMPLATES"), main_app.templates_path -%>
+</li>
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_dropdown_template_admin.html.erb]: Do we really need a partial for 3 lines of a link that's used in just one place? Remember that partials have a rendering overhead.

File: server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_dropdown_pipeline_group_admin.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<li role="presentation">
+  <%= link_to l.string("PIPELINES_POPUP_TITLE"), main_app.pipeline_groups_path -%>
+</li>
+<%- if is_user_a_template_admin? -%>
+  <%= render :partial => 'navigation_elements/dropdown_template_admin.html' -%>
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_dropdown_pipeline_group_admin.html.erb]: There's no need for the `.html` suffix on the partial name.

File: server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_navigation.html.erb [hunk 5]
```diff
@@ -1,41 +1,57 @@
...
-              <li><a href="#">OAuth Enabled Gadget Providers</a></li>
-              <li><a href="#">Backup</a></li>
-              <li><a href="#">Plugins</a></li>
-              <li><a href="#">Package Repositories</a></li>
-          </ul>
-      </li>
-    </ul>
+      </ul>
+      <ul class="left header-tabs">
+        <li id="pipelines-nav-tab" class="<%= "current" if @current_tab_name == "pipelines" %>">
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_navigation.html.erb]: Who sets `@current_tab_name`?

File: server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_dropdown_admin.html.erb [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+</li>
+<%= render :partial => 'navigation_elements/dropdown_template_admin.html' -%>
+<li role="presentation">
+  <%= link_to l.string("PIPELINES_SNIPPET"), main_app.config_view_path -%>
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_dropdown_admin.html.erb]: Please drop any `l.string` in the view. There's no need for any partial internationalization.

File: server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_footer.html.erb [hunk 7]
```diff
@@ -9,15 +9,17 @@
...
             <span class="inline-list social">
                 <a href="http://twitter.com/goforcd" class="twitter"></a>
                 <a href="https://github.com/gocd/gocd" class="github"></a>
                 <a href="https://groups.google.com/d/forum/go-cd" class="forums"></a>
                 <a href="https://go.cd/current/documentation" class="documentation"></a>
-                <a href="https://go.cd/current/cctray" class="plugins"></a>
-                <a href="https://go.cd/current/documentation" class="cctray"></a>
+                <a href="http://www.go.cd/community/plugins.html" class="plugins"></a>
+                <a href="https://api.go.cd" class="api"></a>
+                <a href="<%= url_for_path('about') %>" class="server-details"></a>
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_footer.html.erb]: Did you mean `about_path` and `cctray_path`?

File: server/webapp/WEB-INF/rails.new/app/assets/stylesheets/pipeline_configs/_footer.scss [hunk 8]
```diff
@@ -56,9 +56,15 @@
     &.cctray {
       @include footer-icon($icon: rss);
     }
+    &.api {
+      @include footer-icon($icon: gears);
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/assets/stylesheets/pipeline_configs/_footer.scss]: This a better icon http://fontawesome.io/icon/code/?

File: server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_navigation.html.erb (hunk 1 above)
Comment (varshavaradarajan) [inline server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_navigation.html.erb]: The cas_errors_controller sets no_tabs to true.

File: server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_footer.html.erb (hunk 7 above)
Comment (varshavaradarajan) [inline server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_footer.html.erb]: Yes, we'll change that.

File: server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_navigation.html.erb (hunk 5 above)
Comment (varshavaradarajan) [inline server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_navigation.html.erb]: Each of the tabs' controllers set @current_tab_name.

File: server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_navigation.html.erb (hunk 1 above)
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_navigation.html.erb]: AFAIK — this new set of view partials is only used on the new pipeline config edit SPA, no other controller(s) use it at this time.

You seem to have copied it from existing view partials, that seem to be used in a lot more than one place, so there's some view logic in there to deal with that.

File: server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_navigation.html.erb (hunk 5 above)
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_navigation.html.erb]: See comment above (https://github.com/gocd/gocd/pull/1816/files#r50630024)

File: server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_navigation.html.erb [hunk 9]
```diff
@@ -1,40 +1,56 @@
...
+      <% end %>
+      <li>
+        <%= link_to("Help", "http://www.go.cd/documentation/user/current", :target => "_blank") %>
+      </li>
     </ul>
-    <ul class="left">
-      <li class="active" ><a href="#">Pipeline</a></li>
-      <li><a href="#">Environments</a></li>
-      <li><a href="#">Agents</a></li>
-      <li class="has-dropdown not-click">
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_navigation.html.erb]: If I'm not mistaken that 'not-click' is needed for some hovering/clicking issues. Can you check?

Comment (juhijariwala) [inline server/webapp/WEB-INF/rails.new/app/views/navigation_elements/_navigation.html.erb]: We checked it. It doesn't make any difference. The ```"topbar"``` class adds the ```"not-click"``` class.

**Comment by zabil** [comment]
@juhijariwala has to put in one more fix (broken menu on smaller screens) before this gets merged.

**Comment by juhijariwala** [comment]
I have pushed that fix. Before you merge, I want to run PR build for this.

On Wed, Feb 3, 2016 at 3:07 PM, Zabil C.M <notifications@github.com> wrote:

> @juhijariwala <https://github.com/juhijariwala> has to put in one more
> fix (broken menu on smaller screens) before this gets merged.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gocd/gocd/pull/1816#issuecomment-179124861>.
>



-- 
Cheers,
Juhi Jariwala

**Comment by ketan** [comment]
> I have pushed that fix.

Could you elaborate what that fix was?

**Comment by juhijariwala** [comment]
We removed this Menu
<https://github.com/gocd/gocd/pull/1816/files#diff-d060a52df823d337d1dcc450347dac44R6>
thinking
it is not being used. Then we realised that it is for Mobile view. We ran
all the tests locally since the PR build was failing due jdk8 issue. You
can merge it.

On Fri, Feb 5, 2016 at 10:10 AM, Ketan Padegaonkar <notifications@github.com
> wrote:

> I have pushed that fix.
>
> Could you elaborate what that fix was?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gocd/gocd/pull/1816#issuecomment-180194341>.
>



-- 
Cheers,
Juhi Jariwala

**Pull request merged**